internal/pkg/service/stream/storage/repository/repository.go

```diff
@@ -1,60 +1,60 @@
 package repository
 
 import (
-	"github.com/benbjohnson/clock"
-	"github.com/keboola/go-client/pkg/keboola"
 	etcd "go.etcd.io/etcd/client/v3"
 
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/serde"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/servicectx"
 	definitionRepo "github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition/repository"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/plugin"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/model"
-	statsRepo "github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/statistics/repository"
+	file "github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/repository/file"
+	slice "github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/repository/slice"
+	volume "github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/repository/volume"
 )
 
 type dependencies interface {
-	Clock() clock.Clock
+	Logger() log.Logger
+	Process() *servicectx.Process
 	EtcdClient() *etcd.Client
 	EtcdSerde() *serde.Serde
-	KeboolaPublicAPI() *keboola.PublicAPI
+	Plugins() *plugin.Plugins
 	DefinitionRepository() *definitionRepo.Repository
-	StatisticsRepository() *statsRepo.Repository
 }
 
 // Repository provides database operations with the storage entities.
-// The orchestration of these database operations with other parts of the platform is handled by an upper facade.
 type Repository struct {
-	hook   *hook
-	sink   *definitionRepo.SinkRepository
-	file   *FileRepository
-	slice  *SliceRepository
-	token  *TokenRepository
-	volume *VolumeRepository
+	volume *volume.Repository
+	file   *file.Repository
+	slice  *slice.Repository
 }
```

Comment:
Previously, the token entity was part of the storage repository, but now it is solved through the plugin system.